src/backend/commands/createas.c

```diff
@@ -418,6 +420,18 @@ ExecCreateTableAs(CreateTableAsStmt *stmt, const char *queryString,
 								GetActiveSnapshot(), InvalidSnapshot,
 								dest, params, 0);
 
+	if (gp_enable_gpperfmon 
+			&& Gp_role == GP_ROLE_DISPATCH 
+			&& log_min_messages < DEBUG4)
```

Comment:
1. Indeed, such change solves both problems: empty submit time and presence with default log level. BUT! This patch affects not only queries under plpgsql functions, but any CTAS.
2. Is it the only place when perfmon packet isn't initialized properly?

Reply:
Seems like for pl/pgsql queries initialization is done on a different path and correct perfmon record isn't produuced